modules/zenith-public/cpp/fishing_fatigue.cpp

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------
+-- Fishing Fatigue
+---------------------------------*/
+#include "map/packets/chat_message.h"
+#include "map/utils/charutils.h"
+#include "map/utils/moduleutils.h"
+
+extern uint8                                                                     PacketSize[512];
+extern std::function<void(MapSession* const, CCharEntity* const, CBasicPacket&)> PacketParser[512];
+
+class FishingFatigueModule : public CPPModule
+{
+    std::string CAUGHT = "[FISHING]CAUGHT";
+
+    const uint8  SANDORIA_NATION_ID = 0;
+    const uint8  BASTOK_NATION_ID   = 1;
+    const uint8  WINDURST_NATION_ID = 2;
+    const uint8  RANK_REQUIRED      = 3;
+    const uint16 FISHING_LOWER      = 20;  // Lower limit for fishing
+    const uint16 FISHING_UPPER      = 200; // Upper limit for fishing
+
+    uint16 getCaughtLimit(CCharEntity* PChar)
+    {
+        uint8 rankSandoria           = PChar->profile.rank[SANDORIA_NATION_ID];
+        uint8 rankBastok             = PChar->profile.rank[BASTOK_NATION_ID];
+        uint8 rankWindurst           = PChar->profile.rank[WINDURST_NATION_ID];
```

Comment:
```suggestion
        uint8 rankSandoria           = PChar->profile.rank[NATION_SANDORIA];
        uint8 rankBastok             = PChar->profile.rank[NATION_BASTOK];
        uint8 rankWindurst           = PChar->profile.rank[NATION_WINDURST];
```

May have to include `zone.h` for those enums